sky/utils/atomic.py

```diff
@@ -19,56 +19,35 @@ def get(self) -> int:
         with self._lock:
             return self._value
 
-    def set(self, value: int) -> None:
-        """Set the value atomically.
+    def __iadd__(self, delta: int) -> 'AtomicInt':
+        """Atomically increment by delta and return self.
 
         Args:
-            value: The new integer value to set.
-        """
-        with self._lock:
-            self._value = value
-
-    def increment(self, delta: int = 1) -> int:
-        """Atomically increment by delta and return new value.
-
-        Args:
-            delta: Amount to increment by (default: 1)
+            delta: Amount to increment by
 
         Returns:
-            The new value after incrementing.
+            Self reference to support the += operator
         """
         with self._lock:
             self._value += delta
-            return self._value
+            return self
 
-    def decrement(self, delta: int = 1) -> int:
-        """Atomically decrement by delta and return new value.
+    def __isub__(self, delta: int) -> 'AtomicInt':
+        """Atomically decrement by delta and return self.
 
         Args:
-            delta: Amount to decrement by (default: 1)
+            delta: Amount to decrement by
 
         Returns:
-            The new value after decrementing.
+            Self reference to support the -= operator
         """
         with self._lock:
             self._value -= delta
-            return self._value
+            return self
 
-    def compare_and_set(self, expect: int, update: int) -> bool:
-        """Atomically set to update if current value equals expect.
-
-        Args:
-            expect: The expected current value
-            update: The new value to set if current equals expect
-
-        Returns:
-            True if the value was updated, False otherwise.
-        """
-        with self._lock:
-            if self._value == expect:
-                self._value = update
-                return True
-            return False
+    def decrement(self):
+        """Atomically decrement by 1"""
+        self -= 1
```

Comment:
Q: why do we need both this `decrement` and `__isub__`?

Actually, I found `+=` is a bit harder to read and `increment` and `decrement` is more commonly used for atomic int. How about we revert to `increment` and `decrement` only? Sorry for being back and forth 😅

Reply:
`decrement` simplified the syntax when adding callback to future. And no worries! I actually have no strong opinion between `__iadd__` and `increment`, will just rollback to the previous impl😉